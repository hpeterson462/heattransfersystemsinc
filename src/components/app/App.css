$blue: #003366;
$red: #CC0000;
$lightGray: #F9FAFB;
$gray: #E5E5E5;
$gold: #F68D0B;

html, body {
  margin: 0px;
  padding: 0px;
  width: 100%;
  height: 100%;
  scroll-behavior: smooth;
}

body {
  font-family: 'Roboto', sans-serif;
  overflow-x: hidden;
  margin: 0px;
  padding: 0px;
  border-radius: 2px;
}

h1 {
  display: flex;
  justify-content: center;
  font-size: 3rem;
  font-family: 'Open Sans', sans-serif;
  letter-spacing: 0.2rem;
}

h2 {
  display: flex;
  justify-content: center;
  padding: 20px;
  padding-bottom: 40px;
  width: 70%;
  margin: auto;
  text-align: center;
  line-height: 2.5rem;
  font-size: 1.2rem;
  font-weight: lighter;
  letter-spacing: 0.2rem;
  color: #003366;
  font-family: 'Noto Sans', sans-serif;
}

.company-title {
  text-align: center;
  font-size: 3rem;
}

.contact-email {
  text-decoration: none;
  color: #003366;
}

.contact-email:hover {
  cursor: pointer;
  color: #F68D0B;
}

@media screen and 
  (prefers-reduced-motion: reduce),
  (update: slow) {
    * {
      animation-duration: 0.001ms !important;
      transition-duration: 0.001ms !important;
    }
  }

@media only screen
  and (min-device-width : 20em) 
  and (max-device-width : 50em)
  and (-webkit-min-device-pixel-ratio: 3)
  and (orientation: portrait) {
    html, body {
      overflow-x: hidden;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
           box-sizing: border-box;
    }
}
