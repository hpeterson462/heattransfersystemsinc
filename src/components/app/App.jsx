import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
import Home from '../home/Home';
import Services from '../servicesPage/Services';
import Solutions from '../solutions/Solutions';
import './App.css';

export default function App() {
  return (
    <div>
      <Router>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/solutions" component={Solutions} />
          <Route exact path="/services" component={Services} />
          {/* <Route exact path="/products" component={Products} />
          <Route exact path="/aboutus" component={AboutUs} />
          <Route exact path="/contact" component={Contact} /> */}
        </Switch>
      </Router>
    </div>
  );
}
