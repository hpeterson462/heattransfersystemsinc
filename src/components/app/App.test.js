import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { Router, Link, MemoryRouter } from "react-router-dom";
import { createMemoryHistory } from "history";
import App from './App';

describe('App', () => {
  afterEach(() => cleanup());

  it('renders App component', () => {
    const { asFragment } = render(<MemoryRouter><App /></MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('links to correct page', () => {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    const pathResult = render(<Router history={history}>
      <Link to={{ path: '/' }}>Home</Link>
    </Router>);

    fireEvent.click(pathResult.getByText('Home'));

    expect(history.location.path).toBe('/');
  });
});
