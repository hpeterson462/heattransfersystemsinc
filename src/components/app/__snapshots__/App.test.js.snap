// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
